cli/src/pcluster/config/cluster_config.py

```diff
@@ -65,49 +70,279 @@ def __repr__(self):
+
+
+class BaseNetworkingConfig(Config):
+    """Represent the networking configuration shared by head node and compute node."""
 
     def __init__(
         self,
-        subnet_id: str,
-        elastic_ip: str = None,
         assign_public_ip: str = None,
         security_groups: List[str] = None,
         additional_security_groups: List[str] = None,
+        proxy: ProxyConfig = None,
     ):
         super().__init__()
-        self.subnet_id = subnet_id
-        self.elastic_ip = elastic_ip
         self.assign_public_ip = assign_public_ip
         self.security_groups = security_groups
         self.additional_security_groups = additional_security_groups
+        self.proxy = proxy
 
 
-class QueueNetworkingConfig(Config):
-    """Represent the networking configuration for the Queue."""
+class HeadNodeNetworkingConfig(BaseNetworkingConfig):
+    """Represent the networking configuration for the head node."""
+
+    def __init__(self, subnet_id: str, elastic_ip: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self.subnet_id = subnet_id
+        self.elastic_ip = elastic_ip
 
-    def __init__(self, subnet_ids: List[str]):
+
+class PlacementGroupConfig(Config):
+    """Represent the placement group for the Queue networking."""
+
+    def __init__(self, enabled: bool = None, id: str = None):
         super().__init__()
+        if enabled is None:
+            enabled = MarkedBool(False)
+        self.enabled = enabled
+        self.id = id
+
+
+class QueueNetworkingConfig(BaseNetworkingConfig):
+    """Represent the networking configuration for the Queue."""
+
+    def __init__(self, subnet_ids: List[str], placement_group: PlacementGroupConfig = None, **kwargs):
+        super().__init__(**kwargs)
         self.subnet_ids = subnet_ids
+        self.placement_group = placement_group
 
 
 class SshConfig(Config):
     """Represent the SSH configuration for a node (or the entire cluster)."""
 
-    def __init__(self, key_name: str):
+    def __init__(self, key_name: str, allowed_ips: str = None):
         super().__init__()
+        if allowed_ips is None:
+            allowed_ips = MarkedStr(CIDR_ALL_IPS)
         self.key_name = key_name
+        self.allowed_ips = allowed_ips
+
+
+class DcvConfig(Config):
+    """Represent the DCV configuration."""
+
+    def __init__(self, enabled: bool, port: int = None, allowed_ips: str = None):
+        super().__init__()
+        if port is None:
+            port = MarkedInt(8843)
+        if allowed_ips is None:
+            allowed_ips = MarkedStr(CIDR_ALL_IPS)
+        self.enabled = enabled
+        self.port = port
+        self.allowed_ips = allowed_ips
+
+
+# ---------------------- Nodes ---------------------- #
+class ImageConfig(Config):
+    """Represent the configuration of an Image."""
+
+    def __init__(self, os: str, custom_ami: str = None):
+        super().__init__()
+        self.os = os
+        self.custom_ami = custom_ami
+        self._validators = []
```

Comment:
this initialization  is already done in the BaseConfig you can avoid it if there are no validators to register.

Reply:
Done